### PR TITLE
Add Django 1.11 to the test matrix

### PR DESCRIPTION
**Pull request by jwalgran**
The updates to .travis.yml were made by running the command as directed in the comment.

I corrected a failing `pylint` check by removing an extraneous `else` from the module.

### CODE REVIEW

**Comment by bittner** [comment]
Thanks for this PR! I'll add a few comments.

File: tox.ini [hunk 1]
```diff
@@ -15,13 +15,14 @@ DJANGO_SETTINGS_MODULE = apptemplates.test.settings
 [tox]
 envlist =
     flake8-py{26,27,32,33,34,35}
-    pylint-py{27,35}-django110
+    pylint-py{27,35}-django{110,111}
```
Comment (bittner) [inline tox.ini]: I think it's sufficient to stick to the latest version, e.g.
``` bash
pylint-py{27,36}-django111
```
This declaration serves only to run Pylint against version 2 and 3 of Python (with any, e.g. the latest, Django version installed).

File: tox.ini [hunk 2]
```diff
@@ -15,13 +15,14 @@ DJANGO_SETTINGS_MODULE = apptemplates.test.settings
...
     py{26,27}-django14
     py{26,27,32,33}-django{15,16}
     py{27,32,33,34}-django17
     py{27,32,33,34,35}-django18
     py{27,34,35}-django19
     py{27,34,35}-django110
+    py{27,34,35}-django111
```
Comment (bittner) [inline tox.ini]: [Django 1.11](https://pypi.python.org/pypi/Django/1.11.3) also supports Python 3.6, let's add this too.

While you're at it, can you also use variable expansion for the Django 1.9 and 1.10 declaration? e.g.
``` bash
py{27,34,35}-django{19,110}
py{27,34,35,36}-django111
```

File: tox.ini [hunk 3]
```diff
@@ -77,3 +79,15 @@ commands = pylint --rcfile=tox.ini apptemplates
...
+commands = pylint --rcfile=tox.ini apptemplates
+deps =
+    django111: Django>=1.11,<2.0
+    pylint
+
+[testenv:pylint-py35-django111]
+commands = pylint --rcfile=tox.ini apptemplates
+deps =
+    django111: Django>=1.11,<2.0
+    pylint
```
Comment (bittner) [inline tox.ini]: As per my comment above, please remove the sections `[testenv:pylint-py27-django110]` and `[testenv:pylint-py35-django110]`, and update the ones you introduced to Python 3.6.

I hope Pylint is happy with Python 3.6, by the way.  :hushed:

**Review by bittner — CHANGES_REQUESTED** [review]
For Python 3.5 and 3.6 it may be required to declare the distro and sudo (see [related issue](https://github.com/travis-ci/travis-ci/issues/4990#issuecomment-310824690)). We'll see once Travis complains or not.

File: .travis.yml [hunk 4]
```diff
@@ -31,10 +30,15 @@ env:
...
   - TOXENV=py34-django19
-  - TOXENV=py34-django110
+  - TOXENV=py35-django110
+  - TOXENV=py35-django111
   - TOXENV=py35-django18
   - TOXENV=py35-django19
-  - TOXENV=py35-django110
+  - TOXENV=py36-django111
+  - TOXENV=pylint-py27-django111
+  - TOXENV=pylint-py35-django111
```
Comment (bittner) [inline .travis.yml]: I know what the script does is one thing, but ... can you make a true version sort, i.e. Django 1.8 < 1.9 < 1.10 < 1.11 grouped by Python 2.7 < 3.3 < 3.4 < 3.5 < 3.6?
```
...
- TOXENV=py34-django17
- TOXENV=py34-django18
- TOXENV=py34-django19
- TOXENV=py34-django110  # must be placed here manually
- TOXENV=py34-django111  # must be placed here manually, too
- TOXENV=py35-django18
- TOXENV=py35-django19
- TOXENV=py35-django110  # same as above
- TOXENV=py35-django111  # same as above, again
...
- TOXENV=py36-django1xx  # Django 1.8, 1.9, 1.10 should have been added by the script
...
- TOXENV=py36-django111
```
Thanks for your understanding!

**Review by bittner — CHANGES_REQUESTED** [review]

**Comment by jwalgran** [comment]
After switching to the Travis `trusty` distribution, the 2.6 builds started failing.
```
ERROR:   flake8-py26: InterpreterNotFound: python2.6
```
Do you want to drop 2.6 from the build matrix?

**Comment by jwalgran** [comment]
The Travis docs lead me to believe that testing 2.6 through 3.6 should be possible

https://docs.travis-ci.com/user/languages/python/#Choosing-Python-versions-to-test-against

Trying out adding multiple versions to the `python:` key in `.travis.yml`.

**Comment by bittner** [comment]
> Do you want to drop 2.6 from the build matrix?

If this is the only way to make builds with Python 3.5 and 3.6 pass, yes. It would be nice to keep it though.

The  code base of this package is so simple that it still can run in super-outdated projects (namely the ones still running on Python 2.6).  :smile:

**Comment by bittner** [comment]
> The Travis docs lead me to believe that testing 2.6 through 3.6 should be possible

The challenge is the combination of the Python and Django versions. If you find a way, cool!  If not, don't worry too much, and let's drop 2.6.

**Comment by jwalgran** [comment]
Specifying 2 versions
```
python:
  - 2.6
  - 3.6
```
resulted in Travis running the full matrix twice, so that is not the solution.

https://travis-ci.org/bittner/django-apptemplates/builds/250841808

**Comment by jwalgran** [comment]
Adding this `before_install` section to apt get the missing interpreters did the trick

```
before_install:
  - sudo add-apt-repository -y ppa:fkrull/deadsnakes
  - sudo apt-get -qq update
  - sudo apt-get -y install python2.6 python3.2

File: tox.ini (hunk 3 above)
Comment (jwalgran) [reply tox.ini]: Done. Pylint is happy :-)

**Review by bittner — APPROVED** [review]

**Comment by bittner** [comment]
Thanks a million for your efforts, @jwalgran!

**Pull request merged**